docs/architecture_design.md

```diff
@@ -82,7 +82,16 @@ If the damp score goes down and then decays, the problem is fixed and it will no
 Say the damp score for B exceeds the red line. A fans out a damp-req request to _k_ random members and asks for their damp score of B. If they also communicate that B is flapping, then B is considered damped due to excessive flapping. A marks B as damped, and disseminates that information using the gossip protocol.
 
 ### Full Syncing
-Content coming soon...
+When a node *a* contacts another node *b*, node *a* sends the checksum of its membership. Node *b* compares the checksums and if they mismatch and there are no more changes to be disseminated, node *b* adds its full membership information in the response. This is called a full sync.
+
+One problem with this technique is that node *a* has not sent its membership information to *b* yet and in some cases this prevents the ring to converge. To make the ring in those cases converge, we need bidirectional full syncs -- a way for *b* to get *a*'s full membership as well.
+
+Bidirectional full syncs work as follows. When *b* issues a full sync, *b* also want to know about *a*'s membership. To achieve this *b* sends a join request to *a* and applies the join response to its membership.
+
+One example of a cluster configuration that would not converge with only full syncs:
+Partition A in which every node has A⋃B as alive in its membership;
+Partition B in which every node has B as alive in its membership.
+When node *a* of partition A contacts node *b* of partition B; *b* finds that the checksums do not match. Therefor a full sync is issued and *b* collects its entire membership for *a*. However, *a* already knows about all the members of *b* and no changes are applied and the ring doesn't converge.
```

Comment:
Also, write a **bottom line**, e.g.: *therefore, both clusters will have different notions of the membership, and will never converge.*